lib/resty/cassandra/policies/lb/req_dc_rr.lua

```diff
@@ -47,7 +51,11 @@ function _M:init(peers)
 
   for i = 1, #peers do
     if type(peers[i].data_center) ~= 'string' then
-      error('peer '..peers[i].host..' data_center field must be a string')
+      log(WARN, _log_prefix, 'peer ', peers[i].host, ' data_center field must be a string')
+
+      if peers[i].err ~= '' then
+        log(WARN, _log_prefix, 'peer ', peers[i].host, ' ', peers[i].err)
+      end
     end
 
     if peers[i].data_center == self.local_dc then
```

Comment:
I think this PR isn't updating the most crucial line of code in this error path, this one. The goal is to aim at making it robust in all cases so that if and only if the data center is specified as a string *and* equal to the local data center's name, then and only then do we consider it local.